model/modelImpl.go

```diff
@@ -2,31 +2,76 @@ package model
 
 import (
 	"doccer/auth"
+	"doccer/data"
+	"doccer/linter"
 	"github.com/dgrijalva/jwt-go"
 	"time"
 )
 
 type ModelImpl struct {
 	storage Storage
 	jwtHandler auth.JwtHandler
+	processChannel chan data.Doc
+	resChannel chan data.Doc
 }
 
-func NewModelImpl(storage Storage, secret []byte) ModelImpl {
-	return ModelImpl{
+func NewModelImpl(
+	storage Storage,
+	secret []byte,
+	linter linter.GeneralLinter,
+    saveWorkersCnt int,
+	linterWorkersCnt int,
+	) ModelImpl {
+
+	processChannel := make(chan data.Doc, linterWorkersCnt * 2)
+	resChannel := make(chan data.Doc, saveWorkersCnt * 2)
+
+	res := ModelImpl{
 		storage: storage,
 		jwtHandler: auth.NewJwtHandler(secret, 24 * time.Hour),
+		processChannel: processChannel,
+		resChannel: resChannel,
+	}
+
+	for i := 0; i < saveWorkersCnt; i++ {
+		go func() {
+			for {
+				newDoc, ok := <-resChannel
+				if !ok {
+					break
+				}
+				_, err := res.editDoc(newDoc.AuthorId, newDoc, false)
+				if err != ErrNotFound {
```

Comment:
What is going on here? I can't understand...(

Reply:
With error? Retry if unknown error was returned

Reply:
Is it ok to have just 2 retires? Is it ok to just give up on the second retry? Why don't you making retries in other places?

Reply:
Because other requests could be retried by client. And currently I don't know any reason why should we retry request if it fails and user has access to requested data and that data exists. This probably means that database is not connected. I removed retry there and added endpoint to launch linter